index.js

```diff
@@ -1,68 +1,65 @@
-    const email = document.createElement('p');
-    email.className = 'email'
-    email.innerText = monsterDiv.email;
-    monster.append(img,mname,email);
-    document.querySelector('.monsters').append(monster);
-    console.log(monster);
-}
+function showMonster(monsterDiv) {
+  const monster = document.createElement('div');
+  monster.className = 'monster';
 
+  const img = document.createElement('img');
+  img.src = `https://robohash.org/${monsterDiv.id}?set=set2`;
+  img.alt = monsterDiv.name;
 
-function showMonsterNotFound() {
-    const notFound= document.createElement('div');
-    notFound.className='p-5 not-found';
-    // notFound.style.display='none';
-    const span = document.createElement('span');
-    span.innerText='404';
-    const h1 = document.createElement('h1')
-    h1.innerText='🧟‍♂️ No Monster Found 🧟‍♂️' ;
-    notFound.append(span,h1);
-    document.querySelector('.monsters').append(notFound);
-    console.log(notFound);
+  const mname = document.createElement('p');
+  mname.className = 'name';
+  mname.innerText = monsterDiv.name;
 
+  const email = document.createElement('p');
+  email.className = 'email';
+  email.innerText = monsterDiv.email;
+  monster.append(img, mname, email);
 
+  return monster;
 }
-showMonsterNotFound()
 
-document.querySelector('#search-monster').addEventListener('keyup', function(e) {
-    const keyword= e.target.value.toLowerCase();
-    const findMonster = document.querySelectorAll('.monster');
-    let notFound = true;
+function showMonsterNotFound() {
+  const notFound = document.createElement('div');
+  notFound.className = 'p-5 not-found';
+  const span = document.createElement('span');
+  span.innerText = '404';
+  const h1 = document.createElement('h1');
+  h1.innerText = '🧟‍♂️ No Monster Found 🧟‍♂️';
+  notFound.append(span, h1);
+  return notFound;
+}
 
+monsterFilter.addEventListener('input', function (e) {
+  if (e.target.value !== '') {
+    const keyword = e.target.value.toLowerCase();
 
-    for(let monster of findMonster){
-        const name =monster.children[1].innerText.toLowerCase();
-        const email = monster.children[2].innerText.toLowerCase();
-        if(name.includes(keyword) || email.includes(keyword)){
-            monster.style.display = 'block';
-            notFound=false;
+    const filtered = monsters.filter((el) => {
+      if (el.name.toLowerCase().includes(keyword) || el.email.toLowerCase().includes(keyword)) {
+        return true;
+      }
+    });
 
-        }
-        else{
-            monster.style.display = 'none';
-        }
-    }
-    if(notFound){
-        document.querySelector('.not-found').style.display ='block' ;
-    }
-    else{
-        document.querySelector('.not-found').style.display ='none' ;
+    if (filtered.length === 0) {
+      monsterContainer.innerHTML = '';
+      monsterContainer.append(showMonsterNotFound());
+    } else {
+      populateMonsters(filtered);
     }
-})
+  }
+});
 
+populateMonsters(monsters);
```

Comment:
Instead of the for..of loop which creates the monsters on page load, we're now repurposing the populateMonsters() function with the monsters data set.